ddtrace/tracer/payload.go

```diff
@@ -108,9 +109,22 @@ func (p *payload) updateHeader() {
 func (p *payload) Read(b []byte) (n int, err error) {
 	if p.off < len(p.header) {
 		// reading header
-		n = copy(b, p.header[p.off:])
+		n, err := p.buf.Write(p.header[p.off:])
+		if err != nil {
+			return n, err
+		}
 		p.off += n
-		return n, nil
 	}
+
+	// fill buffer with msgpack-encoded traces, popping them  successively
```

Comment:
When commenting, I've generally used the rule of thumb that the comment should be where the action happens, in this case, inside the `for` loop on the first line, not outside. I think a very simple comment should suffice, as I initially suggested, simply:
```go
// fill buffer
```
Or no comment at all.

Reply:
Done.